src/utils/helpers/focus-trap/focus-trap.js

```diff
@@ -0,0 +1,62 @@
+export default class FocusTrap {
+  constructor(element) {
+    this.element = element;
+    this.focusableElements = undefined;
+    this.firstFocusableElement = undefined;
+    this.lastFocusableElement = undefined;
+    // eslint-disable-next-line max-len
+    this.focusableSelectors = 'button, [href], input:not([type="hidden"]), select, textarea, [tabindex]:not([tabindex="-1"])';
+  }
+
+  blockTabbing(ev) {
+    if (ev.keyCode === 9) {
+      ev.preventDefault();
+    }
+  }
+
+  setFocusableItems() {
+    this.focusableElements = this.element.querySelectorAll(this.focusableSelectors);
+    this.firstFocusableElement = this.focusableElements[0];
+    this.lastFocusableElement = this.focusableElements[this.focusableElements.length - 1];
+  }
+
+  setFocusToFirstFocusableElement() {
```

Comment:
This breaks the `autoFocus` functionality provided by input components. 
E.g if you have a dialog, with a `Textbox` with `autoFocus` set, it will focus the close button instead of the `TextBox`.

Reply:
You are right but new behavior is done because of the new acceptance criteria of this component
```When a dialog opens, place focus on the first focussable element within it.```